src/solc_0.5/Land/erc721/LandBaseTokenV2.sol

```diff
@@ -303,6 +303,73 @@ contract LandBaseTokenV2 is ERC721BaseTokenV2 {
         }
     }
 
+    function exists(uint256 _id) public view returns(bool){
+        uint256 quadType = uint8(_id >> 248);
+        if(quadType == 0){
+            return _exists1x1(_id);
+        }else if(quadType == 1){
+            require(getX(_id)%3 == 0 && getY(_id)%3 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists3x3(_id);
+        }else if(quadType == 2){
+            require(getX(_id)%6 == 0 && getY(_id)%6 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists6x6(_id);
+        }else if(quadType == 3){
+            require(getX(_id)%12 == 0 && getY(_id)%12 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists12x12(_id);
+        }else if(quadType == 4){
+            require(getX(_id)%24 == 0 && getY(_id)%24 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists24x24(_id);
+        }
+        revert("LandBaseTokenV2: Invalid Id");
+    }
+
+    function getX(uint256 _id) internal pure returns(uint256){
+        return ((_id << 8 )>> 8)%GRID_SIZE;
+    }
+
+    function getY(uint256 _id) internal pure returns(uint256){
+        return ((_id << 8 )>> 8)/GRID_SIZE;
+    }
+    function _exists1x1(uint256 _id) internal view returns(bool){
```

Comment:
can u do only one function to handle that ? 

like the function `_ownerOfQuad`